server/controllers/todo.js

```diff
@@ -0,0 +1,80 @@
+const Todo = require("../models").Todo;
+const TodoItem = require("../models").TodoItem;
+
+module.exports = {
+  createTodo(req, res) {
+    return Todo
+    .create({
+       title: req.body.title
+      })
+    .then(todo => res.status(201).send(todo))
+    .catch(err => res.status(500).send(err));
+  },
+
+  getAllTodos(req, res) {
+    return Todo
+    .findAll({
+      include: [{
+        model: TodoItem,
+        as: 'todoItems',
+      }],
+    })
+    .then(todo => res.status(200).send(todo))
+    .catch(err => res.status(500).send(err));
+  },
+
+  findATodo(req, res) {
+    return Todo
+    .findById(req.params.id, {
+      include: [{
+        model: TodoItem,
+        as: "todoItems"
+      }]
+    })
+    .then(todo => {
+      if(!todo) {
+        res.status(404).send({ message: "Not found"});
+      } else {
+        res.status(200).send(todo);
+      }})
+    .catch(err => {
+      res.status(500).send(err)});
+  },
+
+  updateATodo(req, res) {
+    return Todo
+    .findById(req.params.id, {
+      include: [{
+        model: TodoItem,
+        as: "todoItems"
+      }]
+    })
+    .then(todo => {
+      if(!todo) {
+        res.status(404).send({ message: "Not found"});
+      } else {
+        const newTodo = {
+          title: req.body.title
+        }
+        return todo.update(newTodo)
+        .then(() => res.status(200).send(todo))
+        .catch((error) => res.status(400).send(error));
+      }})
+    .catch(err => {
+      res.status(500).send(err)});
+  },
+
+  deleteATodo(req, res) {
+    return Todo
+    .findById(req.params.id)
+    .then(todo =>{
```

Comment:
This should be refactored. promises implements flatMap. so you should not next .thens

Reply:
Yeah i understand i mean you shouldn't have nexted thens